doc/specs/#140 - ZIP Support.md

```diff
@@ -0,0 +1,90 @@
+
+>Compression Ratio is defined as the sum of the sizes of all the items contained within a ZIP file, divided by the size of the ZIP file.
+
+The check will be as follows:
+
+1. For any archive file, we will start by using Windows Shell APIs which treat ZIP files as a Folder object that can be utilized for viewing the contents (we are not extracting to a specified location, yet). We will iterate through each item and determine the cumulative sum of the sizes of all items contained in the zip and compare that value with the size of the ZIP file. **The check will verify that the compression ratio is no greater than 5:1 (20%)**. Any file that has a larger compression ratio (<20%) will be deemed as "suspicious" and a warning will be displayed to the user. Lossless compression of common data files tend to achieve compression ratios of ~2:1 (50%). Any extraction or copying of files will not take place until this check is passed.
+
+2. If a ZIP archive file contains nested archive files, we will support decompressing up to a maximum of 3 layers (2 additional layers including the initial ZIP file). 
+
+If either of these checks fail, the process will terminate and a warning will be displayed to the user. The user can include `--force` to bypass this check and continue installation at their own risk.
+
+>Note: The compression ratio threshold and the number of supported zip layers is not permanent and can be changed based on community feedback.
+
+> Although ZIP headers can contain information about the decompressed size of the file, they are not reliable sources of information as they can be modified externally. ZIP headers also differ from that of .cab or .tar.gz headers, therefore we should avoid taking a dependency on the data acquired from headers when possible.
+
+## Supporting Nested Portable(s) in an Archive
+Currently, information regarding a single installed portable is stored in the ARP entry. In order to support installing a single or multiple portables contained inside an archive file, an additional table of information will need to be appended to the tracking catalog. The table will contain a list of files that were created and placed down as well as various metadata to enable us to verify whether they have been modified. This table should replace most of the uninstall-related information that is stored in ARP for a given portable package. 
+
+The table will contain the following information for each item that we create or place down during installation.
+
+| Metadata    | Description |
+| ----------- | ----------- |
+| Path      | Path to the item |
+| Flag   | Flag to indicate what type of file was placed down (directory, symlink, exe) |
+| Hash   | Hash for files with contents (exe) |
+| SymlinkTarget | Target exe for the created symlink (only applies to symlink files) |

```

Comment:
nit: are these metadata leftover from previous portable spec? They were not referenced in this spec.

Reply:
No, this is metadata that we intend to capture in our tracking catalog instead of recording them in ARP to support the uninstall scenario for portable(s) contained inside a zip.

Reply:
Not tracking catalog, but a separate record of files stored with the extracted files.

Reply:
Updated description to clarify that this is not appended to the tracking catalog but instead captured as a separate record.